presto-main/src/main/java/com/facebook/presto/sql/planner/PlanOptimizers.java

```diff
@@ -169,7 +169,10 @@ public PlanOptimizers(Metadata metadata, SqlParser sqlParser, FeaturesConfig fea
                 new WindowFilterPushDown(metadata), // This must run after PredicatePushDown and LimitPushDown so that it squashes any successive filter nodes and limits
                 new IterativeOptimizer(
                                 stats,
-                                ImmutableList.of(new MergeWindows()),
+                                ImmutableList.of(
+                                        new UnaliasSymbolReferences(),
+                                        new PruneIdentityProjections(),
+                                        new MergeWindows()),
                                 ImmutableSet.of(
                                         new SwapAdjacentWindowsByPartitionsOrder(),
```

Comment:
Don't you need `RemoveRedundantIdentityProjections` here as well? Unnecessary projections may prevent triggering `SwapAdjacentWindowsByPartitionsOrder`, as well as `MergeAdjacentWindows`.